### DISCUSSION TITLE
Doubt regarding equality of Series and Parallel classes in control module.

### DISCUSSION THREAD

**Discussion by abhiphile** [sympy/sympy#26295] [General]
I was working on control when I found this. 
```
>>> from sympy.abc import s
>>> t1 = TransferFunction(s+2,s,s)
>>> t2 = TransferFunction(s+3,1,s)
>>> Series(t1, t2) == Series(t2, t1)
False
>>> Parallel(t1, t2) == Parallel(t2, t1)
False
```

Shouldn't `Series(t1, t2)` be equal to `Series(t2, t1)` ? Is this the design pattern to be followed in future?

**Comment by Davide-sd** [comment]
`Series` and `Parallel` inherits from `Basic`. So, when you execute `Series(t1, t2) == Series(t2, t1)` you are performing a structural equality, not a mathematical one. What does it mean? With structural comparison, the order of the elements in the `.args` attribute matters.

Usually, sympy's expression exposes the `equals` method in order to perform a mathematical equality. `TransferFunction/Series/Parallel` does not. I guess it could be implemented on some base class of `TransferFunction/Series/Parallel`. Currently (for SISO systems), you can verify if they are mathematical equivalent with this `Series(t1, t2).doit() == Series(t2, t1).doit()`. Note that:

1. `doit()` converts `Series/Parallel` to `TransferFunction`.
2. You are still doing a structural comparison (==), but `TransferFunction` can be seen as the "smallest building block" in the control module. So, in this case it is a mathematical comparison.
